ui/src/app/services/error-handling.service.ts

```diff
@@ -93,4 +90,18 @@ export class ErrorHandlingService {
       });
     });
   }
+
+  // convert error response from API to ErrorResponse object
+  public convertAPIError (e: any) {
+    if (e._body != null && JSON.parse(e._body) != null) {
```

Comment:
Mohs mentioned to me when I was working on the original ErrorResponse that there may be valid error bodies, without following our ErrorResponse bodies, so we should probably handle those somehow.